server/src/main/java/org/elasticsearch/Build.java

```diff
@@ -130,6 +130,21 @@ public static String minimumCompatString(IndexVersion minimumCompatible) {
         }
     }
 
+    /**
+     * Check whether a node version is compatible with the current minimum transport version.
+     * @param version A version identifier as a string
+     * @throws IllegalArgumentException if version is not a valid transport version identifier
+     * @return true if the version is compatible, false otherwise
+     */
+    public static boolean isNodeVersionWireCompatible(String version) {
```

Comment:
Nit: since this is only needed in `NodeMetadata` now, I would move it there; the hope is that we need it only in one place and will go away when we remove/replace version usages there.